### PR TITLE
[WIP] Refactor mount + prepare 4.0

### PR DESCRIPTION
**Pull request by apfelbox**
This PR is now ready. It mainly completely refactored the `mount()` function.

### CODE REVIEW

File: tests/cases/mount/mount.jsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        findOne("#qunit-fixture").innerHTML = `
+            <div id="fixture-1" class="fixture"></div>
+            <div id="fixture-2" class="fixture"></div>
+            <div id="fixture-3" class="fixture"></div>
+        `;
+    },
+});
+
+QUnit.test(
+    "mount() test that the correct parameters are passed, fixes #158",
```
Comment (keichinger) [inline tests/cases/mount/mount.jsx]: The comment `fixes #158` belongs IMO into the commit, not the test case description itself.

```suggestion
    "mount() test that the correct parameters are passed",
```

File: tests/cases/mount/mount.jsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+            init () {}
+        }
+
+        mount(".fixture", ExampleMountComponent, {params: [42], context});
+    }
+);
+
+QUnit.test(
+    "mount() with JSX component",
```
Comment (keichinger) [inline tests/cases/mount/mount.jsx]: Can we get a test-case for a functional JSX component as well?

Comment (apfelbox) [reply tests/cases/mount/mount.jsx]: Was already added.

File: tests/cases/mount/mount.jsx (hunk 1 above)
Comment (apfelbox) [reply tests/cases/mount/mount.jsx]: Yeah, I was thinking about that as well. I just removed it.

**Comment by apfelbox** [comment]
The CI will be fixed in the next PR.

**Comment by keichinger** [comment]
@apfelbox unfortunately the CI has failed. Can we get this fixed in this PR or is this something we fix in an upcoming?

**Comment by apfelbox** [comment]
See the comment above.

**Comment by keichinger** [comment]
Great, GitHub didn't refresh my comments list 😒

**Pull request merged**